### ISSUE TITLE
Error while running command hosts: comparison of Array with Array failed

### ISSUE THREAD

**Issue by all3g** [rapid7/metasploit-framework#5817]
```
msf > hosts 

[-] Error while running command hosts: comparison of Array with Array failed

Call stack:
/opt/metasploit-framework/lib/rex/ui/text/table.rb:197:in `sort!'
/opt/metasploit-framework/lib/rex/ui/text/table.rb:197:in `sort_rows'
/opt/metasploit-framework/lib/rex/ui/text/table.rb:112:in `to_s'
/opt/metasploit-framework/lib/msf/ui/console/command_dispatcher/db.rb:499:in `block in cmd_hosts'
/home/notfound/.rvm/gems/ruby-2.1.6@metasploit-framework/gems/activerecord-4.0.13/lib/active_record/connection_adapters/abstract/connection_pool.rb:294:in `with_connection'
/opt/metasploit-framework/lib/msf/ui/console/command_dispatcher/db.rb:312:in `cmd_hosts'
/opt/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:427:in `run_command'
/opt/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:389:in `block in run_single'
/opt/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:383:in `each'
/opt/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:383:in `run_single'
/opt/metasploit-framework/lib/rex/ui/text/shell.rb:203:in `run'
/opt/metasploit-framework/lib/metasploit/framework/command/console.rb:48:in `start'
/opt/metasploit-framework/lib/metasploit/framework/command/base.rb:82:in `start'
./msfconsole:48:in `<main>'

msf > hosts -h
Usage: hosts [ options ] [addr1 addr2 ...]

OPTIONS:
  -a,--add          Add the hosts instead of searching
  -d,--delete       Delete the hosts instead of searching
  -c <col1,col2>    Only show the given columns (see list below)
  -h,--help         Show this help information
  -u,--up           Only show hosts which are up
  -o <file>         Send output to a file in csv format
  -R,--rhosts       Set RHOSTS from the results of the search
  -S,--search       Search string to filter by
  -i,--info         Change the info of a host
  -n,--name         Change the name of a host
  -m,--comment      Change the comment of a host
  -t,--tag          Add or specify a tag to a range of hosts

Available columns: address, arch, comm, comments, created_at, cred_count, detected_arch, exploit_attempt_count, host_detail_count, info, mac, name, note_count, os_flavor, os_lang, os_name, os_sp, purpose, scope, service_count, state, updated_at, virtual_host, vuln_count, tags

```

**Comment by busterb** [comment]
Hi all3g, it sounds like there is a problem with data in a column somewhere. In your environment, can you try something like these commands:

```
hosts -c address
hosts -c mac
hosts -c name
hosts -c mac
hosts -c os_sp
hosts -c purpose
hosts -c info
hosts -c comments
```

What we want to see is which column might be causing the comparison function to fail when sorting the table for visual representation. If you find a column type that shows the issue, that would lead us closer to solving this.

**Comment by all3g** [comment]
I've clear the error data in order to finish my job. 
So sorry about that. This is the only one record left:

```
msf > hosts -d 

[-] Error while running command hosts: comparison of Array with Array failed

Call stack:
/opt/metasploit-framework/lib/rex/ui/text/table.rb:197:in `sort!'
/opt/metasploit-framework/lib/rex/ui/text/table.rb:197:in `sort_rows'
/opt/metasploit-framework/lib/rex/ui/text/table.rb:112:in `to_s'
/opt/metasploit-framework/lib/msf/ui/console/command_dispatcher/db.rb:499:in `block in cmd_hosts'
/home/notfound/.rvm/gems/ruby-2.1.6@metasploit-framework/gems/activerecord-4.0.13/lib/active_record/connection_adapters/abstract/connection_pool.rb:294:in `with_connection'
/opt/metasploit-framework/lib/msf/ui/console/command_dispatcher/db.rb:312:in `cmd_hosts'
/opt/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:427:in `run_command'
/opt/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:389:in `block in run_single'
/opt/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:383:in `each'
/opt/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:383:in `run_single'
/opt/metasploit-framework/lib/rex/ui/text/shell.rb:203:in `run'
/opt/metasploit-framework/lib/metasploit/framework/command/console.rb:48:in `start'
/opt/metasploit-framework/lib/metasploit/framework/command/base.rb:82:in `start'
/opt/metasploit-framework/msfconsole:48:in `<main>'

msf > hosts 

Hosts
=====

address  mac  name  os_name  os_flavor  os_sp  purpose  info  comments
-------  ---  ----  -------  ---------  -----  -------  ----  --------

```

**Comment by jvazquez-r7** [comment]
Thanks @all3g, as @bcook-r7 points, having the same data you were using to generate the exception would be helpful. Since we haven't I've tried to guess where the problem could be.

After review the callstack and review the `Rex::Ui::Text::Table#sort_rows` method my first bet is which the problem could be due to IPv4 vs IPv6 addresses, since in this case the `<=>` operator will return nil.

In order to reproduce (from master) the next steps can be followed:

```
msf > workspace test_hosts
[*] Workspace: test_hosts
msf > hosts

Hosts
=====

address  mac  name  os_name  os_flavor  os_sp  purpose  info  comments
-------  ---  ----  -------  ---------  -----  -------  ----  --------

msf > hosts -a 2a01:0111:f406:0402:0000:0000:0000:0014
[*] Time: 2015-08-17 19:59:46 UTC Host: host=2a01:111:f406:402::14
msf > hosts

Hosts
=====

address                mac  name  os_name  os_flavor  os_sp  purpose  info  comments
-------                ---  ----  -------  ---------  -----  -------  ----  --------
2a01:111:f406:402::14

msf > hosts -a 172.16.158.1
[*] Time: 2015-08-17 19:59:56 UTC Host: host=172.16.158.1
msf > hosts

[-] Error while running command hosts: comparison of Array with Array failed

Call stack:
/Users/jvazquez/Projects/Code/metasploit-framework/lib/rex/ui/text/table.rb:197:in `sort!'
/Users/jvazquez/Projects/Code/metasploit-framework/lib/rex/ui/text/table.rb:197:in `sort_rows'
/Users/jvazquez/Projects/Code/metasploit-framework/lib/rex/ui/text/table.rb:112:in `to_s'
/Users/jvazquez/Projects/Code/metasploit-framework/lib/msf/ui/console/command_dispatcher/db.rb:499:in `block in cmd_hosts'
/Users/jvazquez/.rvm/gems/ruby-2.1.6@metasploit-framework/gems/activerecord-4.0.13/lib/active_record/connection_adapters/abstract/connection_pool.rb:294:in `with_connection'
/Users/jvazquez/Projects/Code/metasploit-framework/lib/msf/ui/console/command_dispatcher/db.rb:312:in `cmd_hosts'
/Users/jvazquez/Projects/Code/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:427:in `run_command'
/Users/jvazquez/Projects/Code/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:389:in `block in run_single'
/Users/jvazquez/Projects/Code/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:383:in `each'
/Users/jvazquez/Projects/Code/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:383:in `run_single'
/Users/jvazquez/Projects/Code/metasploit-framework/lib/rex/ui/text/shell.rb:203:in `run'
/Users/jvazquez/Projects/Code/metasploit-framework/lib/metasploit/framework/command/console.rb:48:in `start'
/Users/jvazquez/Projects/Code/metasploit-framework/lib/metasploit/framework/command/base.rb:82:in `start'
./msfconsole:48:in `<main>'
```

Assuming this could be the same problem you're finding with your data, I've provided a fix here: https://github.com/rapid7/metasploit-framework/pull/5859

If at some moment you get data which is allowing you to raise the same exception and you can test #5859 it would be helpful :)

Thanks!

**Comment by jvazquez-r7** [comment]
@bcook-r7 already landed #5859 so closing it!
